fixture-monkey-api/src/main/java17/com/navercorp/fixturemonkey/api/generator/SealedTypeObjectPropertyGenerator.java

```diff
@@ -55,4 +57,20 @@ public ObjectProperty generate(ObjectPropertyGeneratorContext context) {
 			childPropertiesByProperty
 		);
 	}
+
+	private static Set<Class<?>> collectPermittedSubclasses(Class<?> type) {
+		Set<Class<?>> subclasses = new HashSet<>();
+		collectPermittedSubclasses(type, subclasses);
+		return subclasses;
+	}
+
+	private static void collectPermittedSubclasses(Class<?> type, Set<Class<?>> subclasses) {
+		if (type.isSealed()) {
+			for (Class<?> subclass : type.getPermittedSubclasses()) {
+				collectPermittedSubclasses(subclass, subclasses);
+			}
+		} else {
+			subclasses.add(type);
+		}
+	}
```

Comment:
I think there is no need to divide it into two methods.

Reply:
If we combine the methods into one, won't that create unnecessary HashSet instance?

Reply:
Ah, I see. I did not see the line 70.

It is better to change the second `collectPermittedSubclasses` to `doCollectPermittedSubclasses` for readability.